src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -235,15 +241,30 @@ private void AnalyzeDirectory(DirectoryInfo directoryPath, ReportFormat reportFo
                         {
                             logger.LogError("\t{failedFile}", failedFile);
                         }
+                        return 6;
                     }
+                    return exitCode;
                 }
             }
             catch (Exception exp)
             {
                 logger.LogError(GetExceptionMessage(exp));
+                return 1;
             }
         }
 
+        private int CalculateExitCode(int exitCode, int res)
+        {
+            if (exitCode == 6 || res == 6)
+                return 6;
+            else if ((exitCode == 5 && res >= 1 && res <= 4) || (res == 5 && exitCode >= 1 && exitCode <= 4))
+                return 6;
+            else if (res == 5)
```

Comment:
Shouldn't this one be `res == 5 || exitCode == 5`? It would also fall into the next `else`, I don't think this case is necessary actually

Reply:
Error: #s 1 -4
Issue: # 5
Error + Issue: # 6

Reply:
`Shouldn't this one be res == 5 || exitCode == 5?` It is implicit. 
Scenario: Exit code is already a 5. a) res is a 0. This function is not called. b) res is 1 - 4. It is covered in above `else if`. c) res is a 5. already covered the way it is written. 
Now for the `I don't think this case is necessary actually`. You are correct. I put it in case we change the end logic. If it stays that way, then I'll remove. Thanks for pointing it out.

Reply:
Yes, but if by the last else either res or exitCode are 5 then that's what's going to be returned, if you do max

Reply:
I think we can remove this function altogether.  See my other comment at the top of this function.